### PR TITLE
Cube cleanups

### PR DESCRIPTION
**Pull request by lbwexler**
+ Rename Record -> CubeRecord
+ Remove additional cruft related to streaming updates.
+ Cleanup documentation, use ES6 collections, iterations

### CODE REVIEW

File: data/cube/View.js [hunk 1]
```diff
@@ -108,26 +107,23 @@ export class View {
...
-                    const lVal = rec.get(f.name);
-                    if (lVal != null && !vals.includes(lVal)) {
-                        vals.push(lVal);
-                    }
-                });
-
-                ret.push({
-                    name: f.name,
-                    displayName: f.displayName,
-                    values: vals
```
Comment (shravnk) [inline data/cube/View.js]: Don't think we're using this right now, but making `values` a Set would change the API, no? Obviously benefit in terms of code cleanup

File: data/cube/Query.js [hunk 2]
```diff
@@ -89,17 +89,8 @@ export class Query {
     // Implementation
     //---------------------------
     parseFields(names) {
-        const cubeFields = this.cube.fields;
```
Comment (shravnk) [inline data/cube/Query.js]: See the benefit of updating to ES6/lodash, just want to confirm that this safety check is unnecessary/unwarranted

File: data/cube/View.js (hunk 1 above)
Comment (lbwexler) [reply data/cube/View.js]: yes -- but we are essentially at a clean slate with the API -- the sencha upgrade is never going to be a dropin at this point.

File: data/cube/Query.js (hunk 2 above)
Comment (lbwexler) [reply data/cube/Query.js]: This was probably me going overboard trying to clean things up.  Agree the safety check is probably worth it.  Will restore post-merge

**Pull request merged**